packages/dds/tree/src/simple-tree/schemaTypes.ts

```diff
@@ -80,6 +81,60 @@ export function isTreeNodeSchemaClass<
  */
 export type AllowedTypes = readonly LazyItem<TreeNodeSchema>[];
 
+/**
+ * Stores annotations for a set of allowed types.
+ * @alpha
+ */
+export interface AnnotatedAllowedTypes {
+	readonly metadata: AllowedTypesMetadata;
+	readonly types: readonly AnnotatedAllowedType[];
+}
+
+/**
+ * Checks if the input is an {@link AnnotatedAllowedTypes}.
+ */
+export function isAnnotatedAllowedTypes(
+	allowedTypes: ImplicitAnnotatedAllowedTypes,
+): allowedTypes is AnnotatedAllowedTypes {
+	return (
+		typeof allowedTypes === "object" && "metadata" in allowedTypes && "types" in allowedTypes
+	);
+}
+
+/**
+ * Annotations that apply to a set of allowed types.
+ * @alpha
+ */
+export interface AllowedTypesMetadata {}
```

Comment:
All types implicitly implement the empty interface, so this isn't very helpful from a static typing perspective. I assume this is because we will be adding options later, cut currently we have none?

If we want to ensure compatibility, we can explicitly document what kinds of changes we are allowed to make to this, we can document something like:

```suggestion
/**
 * Annotations that apply to a set of allowed types.
 * @remarks
 * Additional optional may be added to this as non-breaking changes, so implementations of it should be simple object literals with no unlisted members.
 * @alpha
 */
export interface AllowedTypesMetadata {
custom?: unknown;
}
```

This matches the patterns for some of our other metadata, where we provide a place where users can put anything they want, but reserve the top level object for well know keys.